src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import cn from 'classnames';
+import { useState, useContext } from 'react';
+import { Todo } from '../../services/types';
+import { TodosContext } from '../../TodosContext';
+
+interface Props {
+  todo: Todo,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [titleQuery, setTitleQuery] = useState(todo.title);
+
+  const {
+    handleTodoChange: hanldeTodoChange,
+    handleOnDelete: hanldeOnDelete,
+  } = useContext(TodosContext);
+
+  const handleOnToggle = (isChecked: boolean) => {
+    const newTodo: Todo = {
+      ...todo,
+      completed: isChecked,
+    };
+
+    hanldeTodoChange(newTodo);
```

Comment:
for toggle some todo, you can pass only todoId that you want to switch and handle it like
```
const handleToggleTodo = (todoId: number) => {
  setTodos(prev => prev.map(todo => todo.id === todoId ? { ...todo, completed: !todo.completed } : todo))
}
```

and you don't need to create some newTodo, pass it to the context, have some isChecked argument etc